RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferFastAggregation.java

```diff
@@ -79,11 +79,7 @@ public static MutableRoaringBitmap and(Iterator<? extends ImmutableRoaringBitmap
   public static MutableRoaringBitmap and(long[] aggregationBuffer,
                                          Iterator<? extends ImmutableRoaringBitmap> bitmaps) {
     if (bitmaps.hasNext()) {
-      try {
-        return workShyAnd(aggregationBuffer, bitmaps);
-      } finally {
-        Arrays.fill(aggregationBuffer, 0L);
-      }
+      return workShyAnd(aggregationBuffer, bitmaps);
```

Comment:
Behaviour change

Reply:
@richardstartin What triggers the exception in this instance?

Reply:
You are right, above that the use without clearing buffer does not make sense or it is better to be done at once.

What should be changed I think is to get of rid `finally` section completely as some unexpected exception can occur and then it is ignored or it is also related to GC write barrier? Above that in the case with provided iterator the method returns empty bit map if some exception occurs, which I mean it is not expected behavior. Imagine the use case: roaring bit map used as a ip blacklist and we want match ips from all incoming request with blacklisted ones. The result is no one is on the blacklist when used like `ips.and(buffer, blacklist)`.

What's next is that when the sibling method `RoaringBitmap and(RoaringBitmap... bitmaps) `without provided buffer is called, buffer is uselessly cleared at the end of work even that it is not used at all after, may be it is skipped by JVM optimization, don't know.

Reply:
@richardstartin Can you help us understand this code?

Reply:
> as some unexpected exception can occur and then it is ignored

Finally would not ignore the exception. This code still throws an exception.

https://stackoverflow.com/questions/2614473/does-it-make-sense-to-do-try-finally-without-catch